yjit/src/core.rs

```diff
@@ -429,7 +435,42 @@ fn limit_block_versions(blockid: BlockId, ctx: &Context) -> Context
 // Filled by gen_code_for_exit_from_stub().
 //static uint8_t *code_for_exit_from_stub = NULL;
 
+impl Block {
+    pub fn new() -> Self {
+        let ptr = 0;
+
+        return Block {
+            blockid: BlockId { iseq: IseqPtr(0), idx: 0 },
+            end_idx: 0,
+            ctx: Context::new(),
+            start_addr: CodePtr::from(ptr as *mut u8),
+            end_addr: CodePtr::from(ptr as *mut u8),
+            incoming: Vec::new(),
+            outgoing: Vec::new(),
+            gc_object_offsets: Vec::new(),
+            cme_dependencies: Vec::new(),
+            entry_exit: CodePtr::from(ptr as *mut u8)
+        };
+    }
+}
+
 impl Context {
+    pub fn new_with_size_and_offset(stack_size: u16, sp_offset: i16) -> Self {
```

Comment:
For testing purposes maybe we want a shorthand `new_with_stack_size(n)` ? Would make it clearer what we're doing when reading code.